api/src/exam-environment/utils/exam.ts

```diff
@@ -7,22 +7,33 @@ import {
   EnvGeneratedExam,
   EnvMultipleChoiceQuestion,
   EnvQuestionSet,
-  EnvQuestionSetAttempt,
-  user
+  EnvQuestionSetAttempt
 } from '@prisma/client';
 import { type Static } from '@fastify/type-provider-typebox';
 import * as schemas from '../schemas';
 
+interface CompletedChallengeId {
+  completedChallenges: {
+    id: string;
+  }[];
+}
+
 /**
  * Checks if all exam prerequisites have been met by the user.
- *
- * TODO: This will be done by getting the challenges required from the curriculum.
  */
-export function checkPrerequisites(_user: user, _prerequisites: unknown) {
-  return true;
+export function checkPrerequisites(
+  user: CompletedChallengeId,
+  prerequisites: EnvExam['prerequisites']
+) {
+  return prerequisites.every(p =>
+    user.completedChallenges.find(c => c.id === p)
```

Comment:
```suggestion
    user.completedChallenges.some(c => c.id === p)
```
Slightly more direct.